### PR TITLE
tusc.sh: Experimental support for psitransfer

### PR DESCRIPTION
**Pull request by innovarew**
Just to let you know, that I quickly hacked tusc.sh to support upload to psitransfer.

### CODE REVIEW

**Comment by adhocore** [comment]
thank you for contrib, however I'm not very sure if I should accept the PR

File: tusc.sh [hunk 1]
```diff
@@ -244,12 +253,14 @@ request "-H 'Content-Type: application/offset+octet-stream' \
...
   -H 'Transfer-Encoding: chunked' \
   --upload-file '$FILEPART' \
-  --request PATCH $TUSURL" &
+  --request PATCH $HOST$TUSURL" &
 
 # show spinner
 spinner
 HEADER0=$HEADER HEADER=`mktemp -t tus.XXXXXXXXXX`
 while :; do
+  [[ -z ${HEADERS[Upload-Offset]} ]] && HEADERS[Upload-Offset]=$SIZE
```
Comment (adhocore) [inline tusc.sh]: this makes script immediately exit in line just below, isn't it?

**Comment by adhocore** [comment]
tusc.sh is meant to be generic tus protocol client. if we could integrate psitrans with very little change and impact then maybe we can have it - the current change seems impacting general purpose usecase and common logic as well

File: tusc.sh [hunk 2]
```diff
@@ -244,12 +253,14 @@ request "-H 'Content-Type: application/offset+octet-stream' \
   -H 'Upload-Offset: $OFFSET' \
   -H 'Transfer-Encoding: chunked' \
   --upload-file '$FILEPART' \
-  --request PATCH $TUSURL" &
+  --request PATCH $HOST$TUSURL" &
```
Comment (adhocore) [inline tusc.sh]: will this cause repetition of domain part thus breaking in some cases?

**Comment by innovarew** [comment]
I totally understand the point of not merging the PR as it is now.

I'll be updating the code so the support for psitransfer is less intrusive. The idea is having a flag `tusc.sh --psitransfer` to isolate the psitransfer specific changes inside conditionals if's to not clutter tusc.sh code.

**Comment by adhocore** [comment]
closing for now. pls feel free to open and fix the PR if you so wish

**Pull request closed without merging**